skbio/tree/_tree.py

```diff
@@ -2430,10 +2432,14 @@ def update_result():
             starts, stops = [], []  # to calc ._start and ._stop for curr node
             for child in node.children:
                 if child.length is None:
-                    raise NoLengthError("Node with name '%s' doesn't have a "
-                                        "length." % child.name)
+                    warnings.warn("Node with name '%s' doesn't have a "
+                                  "length." % child.name,
+                                  RepresentationWarning)
 
-                distances[child.__start:child.__stop] += child.length
+                    distances[child.__start:child.__stop] += 0
```

Comment:
Use `0.0` since lengths are internally stored as floats. The current implementation will work but I prefer sticking with a single data type for consistency.